File: IT/L'Indirizzario/Java/Indirizzario Java v3/src/NContact.java
/* file NContact.java */

package programmiamo.ind;

import java.awt.GridLayout;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import javax.swing.BoxLayout;
import javax.swing.JFrame;
import javax.swing.JPanel;
import javax.swing.JTextField;
import javax.swing.JButton;
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.SQLException;

public class NContact extends JFrame {
	private MainFrame parent;

	public NContact(MainFrame parent) {
		this.parent = parent;

		/* crea cinque campi di testo editabile */

		JTextField name = new JTextField("Nome");
		JTextField surname = new JTextField("Cognome");
		JTextField addr = new JTextField("Indirizzo");
		JTextField pnum = new JTextField("Numero di telefono");
		JTextField email = new JTextField("E-mail");

		/* crea il pannello superiore */

		JPanel panel = new JPanel();

		/* imposta il layout manager del pannello intermedio, creando una griglia di tre righe e due colonne */

		panel.setLayout(new GridLayout(3, 2));

		/* aggiunge i campi di testo editabile al pannello superiore */

		panel.add(name);
		panel.add(surname);
		panel.add(addr);
		panel.add(pnum);
		panel.add(email);

		/* crea i pulsanti "Ok" e "Annulla" */

		JButton ok = new JButton("Ok");
		JButton cancel = new JButton("Annulla");

		/* aggiunge i due pulsanti a un nuovo pannello */

		JPanel buttons = new JPanel();

		buttons.add(ok);
		buttons.add(cancel);

		/* gestisce il click sul pulsante "Ok": inserisce i dati del nuovo contatto nel database */

		ok.addActionListener(	new ActionListener() {
			public void actionPerformed(ActionEvent e) 
			{
				/* ottiene un riferimento al MainFrame per accedere ai membri della finestra padre */

				MainFrame parent = NContact.this.parent;
				PreparedStatement pStmt = null;

				try {
					/* aggiorna il database con un preparedStatement */

					pStmt = parent.getConnection().prepareStatement("INSERT INTO contacts (name, surname, addr, pnum, email) VALUES (?, ?, ?, ?, ?)");

					pStmt.setString(1, name.getText());
					pStmt.setString(2, surname.getText());
					pStmt.setString(3, addr.getText());
					pStmt.setString(4, pnum.getText());
					pStmt.setString(5, email.getText());
					pStmt.executeUpdate();

					/* aggiorna la finestra principale */

					parent.updateUI();

				} catch (SQLException s) {
					s.printStackTrace();
				}
				finally {
					try {
						if (pStmt != null) pStmt.close();
					}
					catch (SQLException ex) {
						System.err.println(ex);
					}
				}

				/* chiude la finestra volante */

				NContact.this.dispose();
			}
		});

		/* gestisce il click sul pulsante "Annulla" */
	
		cancel.addActionListener(	new ActionListener() {
			public void actionPerformed(ActionEvent e) {
				NContact.this.dispose();
			}
		});

		this.getContentPane().setLayout(new BoxLayout(this.getContentPane(), BoxLayout.Y_AXIS));

		/* aggiunge i due pannelli alla finestra NContact */

		this.add(panel);
		this.add(buttons);
		this.pack();

		/* chiude la finestra NContact, lasciando aperta la finestra principale dell'applicazione */

		this.setTitle("Inserisci contatto");
		this.setVisible(true);
		this.setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
	}
}

File: IT/2/L'Indirizzario/C++/Indirizzario C++ v2/NContact.cpp
/* file NContact.cpp */

#include <QDialog>
#include <QWidget>
#include <QLabel>
#include <QVBoxLayout>
#include <QGridLayout>
#include <QDialogButtonBox>
#include "MainFrame.h"
#include "NContact.h"

namespace ind = programmiamo::ind;

ind::NContact::NContact(QWidget *parent) : QDialog(parent)
{
	/* crea un pannello contenitore per i campi di testo editabile */

	QWidget* panel = new QWidget(this);

	/* crea cinque campi di testo editabile per inserire il nome, il cognome... */

	this->name = new QLineEdit("Nome", panel);
	this->surname = new QLineEdit("Cognome", panel);
	this->addr = new QLineEdit("Indirizzo", panel);
	this->pnum = new QLineEdit("Numero di telefono", panel);
	this->email = new QLineEdit("E-mail", panel);

	/* crea una griglia per disporre i campi di testo */

	QGridLayout* grLayout = new QGridLayout(panel);

	/* aggiunge le etichette alla griglia */

	grLayout->setMargin(0);
	grLayout->setSpacing(0);
	grLayout->addWidget(name, 0, 0);
	grLayout->addWidget(surname, 0, 1);
	grLayout->addWidget(addr);
	grLayout->addWidget(pnum);
	grLayout->addWidget(email);

	/* associa la griglia al pannello intermedio */

	panel->setLayout(grLayout);

	/* crea un pannello contenitore per i pulsanti */

	QDialogButtonBox *buttons = new QDialogButtonBox(QDialogButtonBox::Ok | QDialogButtonBox::Cancel);

	buttons->setCenterButtons(true);

	/* gestisce il click sul bottone "Ok" */

	connect(buttons, SIGNAL(accepted()), this, SLOT(accept()));

	/* gestisce il click sul bottone "Annulla" */

	connect(buttons, SIGNAL(rejected()), this, SLOT(reject()));

	/* dispone in una striscia verticale la griglia e il pannello con i pulsanti */

	QVBoxLayout *bLayout = new QVBoxLayout(this);

	bLayout->addWidget(panel);
	bLayout->addWidget(buttons);

	this->setLayout(bLayout);

	this->setWindowTitle("Inserisci contatto");
	this->setWindowModality(Qt::ApplicationModal);
	this->setAttribute(Qt::WA_DeleteOnClose);
	this->show();
}

